krisHutchison/answers.md

```diff
@@ -0,0 +1,240 @@
+
+**c)** `x == 'x'`
+
+
+**A:** Error - Because x is undefined it can't be evaluated against string 'x'.
+
+**d)** `x == (x+'')`
+
+
+**A:** Error - again, x is undefined so it cannot be evaluated against (x+'').
+
+**e)** `'' == ' '`
+
+
+**A:** False - Whitespace is included in the second string. They are not equal.
+
+**f)** `x = true`
+
+
+**A:** Defines x as a variable equal to boolean true.
+
+**g)** `var x; x == 'undefined'`
+
+
+**A:** False - Var x evaluates to type undefined which is not equal to string 'undefined'
+
+**h)** `'9'<'10'`
+
+
+**A:** False - strings evaluated as numbers only use the first number so this expression is actually 9 < 1.
+
+**i)** `typeof x + 1 === "number"`
+
+
+**A:** False - because x is undefined it cannot have 1 added to it which makes x + 1 NaN which is not strictly equal to the string 'number'.
+
+**j)** `typeof x % 2 === "number"`
+
+
+**A:** False - again, x % 2 is Nan which is not strictly equal to string 'number'.
+
+**k)** `typeof (x % 2) === "number"`
+
+
+**A:** Error - The parens require x % 2 to be evaluated first which cannot happen because x is undefined.
+
+**l)** `x++ == ++x`
+
+
+**A:** Error - x is undefined and cannot be evaluated.
+
+**m)** `++x == x++`
+
+
+**A:** Error - expression terminates at ++x because x is undefined.
+
+**n)** `"1"+x == 1+x`
+
+
+**A:** Error - expression terminates at "1"+x because x is undefined.
+
+**o)** `"0"+1 == 1`
+
+
+**A:** True - "0"+1 evaluates to string "01" which is converted to a number when compared to number 1. The leading 0 on the string "01" is dropped.
+
+**p)** `(typeof (x+1))===(typeof x)`	
+
+
+**A:** Error - (x+1) is evaluated first but x is undefined so the operation terminates.
+
+**q)** `(x*1 == x) || ((typeof x) != "number")`
+
+
+**A:** Error - the expression is evaluated from within the inner parens. Typeof x is undefined which returns true when evaluated against not string 'number' the operation terminates when it gets to the undefined x on the left side of the expression.
+
+**r)** `(x=(typeof (x+(typeof x))))==x`
+
+
+**A:** Error - the inner (typeof x) evaluates to "undefined" which is added to x leaving a string 'xundefined' which is typeof string but the first x terminates because it's undefined.
+
+---
+
+All of the following can be solved with ordinary expressions and global variables with primitive values.  You don't need functions, loops, or other topics beyond our first two classes.
+
+**2)** (_Difficulty: easy_)
+
+Assume variables x, y, and z are numbers.
+
+**a)**
+Write an expression for the mean (i.e. average) of x, y, and z.
+
+
+**A:** 
+`(x+y+z)/3`
+
+**b)**
+Write a series of expressions to adjust each of x, y, and z halfway toward their mean.
+That is, reset the value of each variable to something based on its previous value.
+
+
+**A:** 
+
+	x = x-(((x-(x+y+z)/3))*0.5)
+	y = ((((x+y+z)/3)-y)*0.5)+y
+    z = ((((x+y+z)/3)-z)*0.5)+z
```

Comment:
You can make your code much easier to read and reduce both duplication and the possibility for error by using variables. For example, if you set `var mean = (x + y + z)/3`, you can then use it in your later equations, like `y = ( mean - y ) * 0.5 + y`